### PR TITLE
Introduce RichTextBlock

### PR DESCRIPTION
**Pull request by Gillibald**
## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021396-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021485-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021540-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021542-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: samples/Sandbox/MainWindow.axaml [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+        with <Span TextDecorations="Underline">several</Span>
+        <Span FontStyle="Italic">Span</Span> elements,
+        <Span Foreground="Blue">
+          using a <Bold>variety</Bold> of <Italic>styles</Italic>
+        </Span>.
+      </RichTextBlock>
+      <TextBox Text="{Binding #txtBlock.SelectedText}"/>
+      <TextBlock Text="{Binding #txtBlock.Text}"/>
+    </StackPanel>
+  </Border>
```
Comment (maxkatz6) [inline samples/Sandbox/MainWindow.axaml:18]:
```suggestion
```

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021680-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**